### PR TITLE
Refactor/add-tests

### PR DESCRIPTION
**Pull request by Somiel**
Add some RSpec unit tests for functionality examination.

### CODE REVIEW

File: spec/factories.rb [hunk 1]
```diff
@@ -0,0 +1,51 @@
+# This will guess the User class factory
+FactoryGirl.define do
```
Comment (murderface13) [inline spec/factories.rb]: you should define one factory in one file. factory :user should be in users_factory.rb, e.g.

File: spec/factories.rb [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  factory :user do
+    first_name 'User'
```
Comment (murderface13) [inline spec/factories.rb]: use Faker to fill table fields by random values, please

File: spec/factories.rb [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+  end
+
+  factory :admin do
+    first_name 'Admin'
+    last_name  'Admin'
+    phone_number 7777777
+    password 'qwerty123'
+    password_confirmation 'qwerty123'
+    email 'admin@example.com'
+    type 'admin'
```
Comment (murderface13) [inline spec/factories.rb]: you should use traits for set special values. see at Traits section here: https://github.com/thoughtbot/factory_girl/blob/master/GETTING_STARTED.md

File: spec/shared/shared_user_behavior.rb [hunk 4]
```diff
@@ -0,0 +1,11 @@
+shared_examples "shared user behavior" do
```
Comment (murderface13) [inline spec/shared/shared_user_behavior.rb]: :+1:

File: spec/factories.rb (hunk 1 above)
Comment (Somiel) [inline spec/factories.rb]: refactored

File: spec/factories.rb (hunk 2 above)
Comment (Somiel) [inline spec/factories.rb]: refactored

File: spec/shared/shared_user_behavior.rb (hunk 4 above)
Comment (Somiel) [inline spec/shared/shared_user_behavior.rb]: :smile:

File: spec/factories.rb (hunk 3 above)
Comment (Somiel) [inline spec/factories.rb]: fixed, take a look please

**Comment by murderface13** [comment]
:+1:

**Pull request merged**